internal/clients/ec2/ec2_client.go

```diff
@@ -34,11 +34,13 @@ func NewEC2Client(ctx context.Context, params ...string) *Client {
 	cache := aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(
 		config.AWS.Key, config.AWS.Secret, config.AWS.Session))
 
-	region := params[0]
-	if region == "" {
-		region = config.AWS.Region
+	region := config.AWS.Region
+	if params != nil {
+		region = params[0]
+		if region == "" {
+			region = config.AWS.Region
+		}
 	}
```

Comment:
This does not fix the bug, variable `params` is never nil, but it can be empty in two out of three cases, therefore `params[0]` panics with out of bounds.

I highly recommend to remove variadic function in favour of two functions: `NewEC2ClientWithRegion` and `NewEC2Client` which will call the former function with config setting. This is the safest option.

Reply:
I am not sure why we need two functions if it will be changed after your PR is merged.
we need to change the jobs as well, each job should hold the region.

Reply:
But I don't mind to change it if you think it is a better option